actionpack/lib/action_dispatch/testing/request_encoder.rb

```diff
@@ -34,7 +34,7 @@ def accept_header
     end
 
     def encode_params(params)
-      @param_encoder.call(params)
+      @param_encoder.call(params) unless params.nil?
```

Comment:
I think it shouldn't be fixed in `actionpack/lib/action_dispatch/testing/integration.rb` See diff:
```diff
diff --git a/actionpack/lib/action_dispatch/testing/integration.rb b/actionpack/lib/action_dispatch/testing/integration.rb
index 7637febd1c..9c6075ad0c 100644
--- a/actionpack/lib/action_dispatch/testing/integration.rb
+++ b/actionpack/lib/action_dispatch/testing/integration.rb
@@ -233,7 +233,7 @@ def process(method, path, params: nil, headers: nil, env: nil, xhr: false, as: n

         request_env = {
           :method => method,
-          :params => request_encoder.encode_params(params),
+          :params => params.nil? ? nil : request_encoder.encode_params(params),
```

Reply:
done

Reply:
@bogdanvlviv why shouldn't it be fixed in here? Isn't it a special case of the request encoder to handle?

I'd write it as `@param_encoder.call(params) if params` though.

Should we update some documentation too? E.g. to say that a request encoder will only be asked to encode params when they aren't nil or similar?

Reply:
Yeah, probably would be better just to write `@param_encoder.call(params) if params` in  `ActionDispatch::RequestEncoder`, Since it is private api we haven't written any documentation.
By the way if

```ruby
get "/foos_json", as: :json, params: false
request.url # => http://www.example.com/foos_json?false
```
condition `if params` would fix it too.

@azbshiri My apologies for confusing. Could you apply `@param_encoder.call(params) if params`, and update the commit message accordance this change?

Reply:
done

Reply:
I didn't commit and push the changes of Yesterday, it's ready to be merged.